test/new-e2e/language-detection/language_detection_test.go

```diff
@@ -0,0 +1,111 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package languagedetection
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+)
+
+// testLocally should be set to true if running the test from a developer machine (i.e. not CI)
+// This sets the test suite to use the hardcoded agent version `versionStr`
+const testLocally = false
```

Comment:
Does it make sense to use an environment variable or an input flag so that this can be run with `go test` without needing to change code? https://blog.jbowen.dev/2019/08/using-go-flags-in-tests/

Reply:
Super interesting ! We're investigating on the best way to split e2e tests from unit tests, considering also build tags https://mickey.dev/posts/go-build-tags-testing/

For this specific case, you can use the env variable `CI`

```golang
import "os"

 isCI, _ := strconv.ParseBool(os.Getenv("CI"))
```

Reply:
That's a good idea. Do you think it would make the workaround for local testing less discoverable though?

Reply:
Done!